### PR TITLE
Task/passport

### PR DESCRIPTION
**Pull request by Abrafaks**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,11 +9,14 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@types/passport-jwt": "3.0.6",
```
Comment (Abrafaks) [inline package.json]: I see it

File: package.json [hunk 2]
```diff
@@ -34,6 +36,7 @@
     "@types/jsonwebtoken": "8.5.3",
     "@types/mongoose": "5.11.97",
     "@types/node": "16.0.0",
+    "@types/passport-jwt": "3.0.6",
```
Comment (MoodupDev) [inline package.json:40]: You can also add `@types/passport`

File: src/services/user.service.ts [hunk 3]
```diff
@@ -53,10 +36,10 @@ export class UserService {
   public async createUserAndReturnToken(
     email: string,
     password: string
-  ): Promise<string> {
+  ): Promise<Object> {
```
Comment (MoodupDev) [inline src/services/user.service.ts:39]: Don't use Object. Define interface. Also don't return token on creating the user

File: src/config/passport.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import { Strategy, ExtractJwt } from "passport-jwt";
+import { User } from "../models/user.model";
+
+const options = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+  secretOrKey: process.env.JWT!,
+  algorithms: ["HS256"],
+};
+
+const strategy = new Strategy(options, async (payload, done) => {
```
Comment (MoodupDev) [inline src/config/passport.ts:11]: I'm not sure that config is a correct place for the strategies. It should be placed in auth module

File: src/controllers/recipe.controller.ts [hunk 5]
```diff
@@ -20,6 +21,8 @@ export class RecipeController {
     try {
       const { title, description, preparing, ingredients, url } =
         req.body as CreateRecipeBody;
+      const { _id } = req.user as UserDocument;
```
Comment (MoodupDev) [inline src/controllers/recipe.controller.ts:24]: Here I'd either change `Request` to `AuthorizedRequest` or some similar interface with defined `user` property or modify globals as we previously did (the structure is a little bit different).

File: src/controllers/user.controller.ts [hunk 6]
```diff
@@ -42,7 +42,7 @@ export class UserController {
           existingUser.isAdmin
         );
 
-        return userService.setCookie(res, token);
+        return res.send({ existingUser, token });
```
Comment (MoodupDev) [inline src/controllers/user.controller.ts:45]: Don't send user together with the token. Separate these logic

**Review by MoodupDev — APPROVED** [review]

**Pull request merged**